test/helpers/kubectl.go

```diff
@@ -1309,3 +1318,104 @@ func (kub *Kubectl) CiliumHealthPreFlightCheck() error {
 	}
 	return nil
 }
+
+// CiliumServicePreFlightCheck checks that k8s service is plumbed correctly
+func (kub *Kubectl) CiliumServicePreFlightCheck() error {
+	svcRes := kub.GetFromAllNS("service")
+	err := svcRes.GetErr("Unable to get k8s services")
+	if err != nil {
+		return err
+	}
+	var k8sSvcs v1.ServiceList
+	err = svcRes.Unmarshal(&k8sSvcs)
+
+	if err != nil {
+		return fmt.Errorf("Unable to unmarshal K8s services: %s", err.Error())
+	}
+
+	epRes := kub.GetFromAllNS("endpoints")
+	err = epRes.GetErr("Unable to get k8s endpoints")
+	if err != nil {
+		return err
+	}
+	var k8sEps v1.EndpointsList
+	err = epRes.Unmarshal(&k8sEps)
+	if err != nil {
+		return fmt.Errorf("Unable to unmarshal K8s endpoints: %s", err.Error())
+	}
+
+	ciliumPods, err := kub.GetCiliumPods(KubeSystemNamespace)
+	if err != nil {
+		return err
+	}
+	ciliumSvcCmd := "cilium service list -o json"
+	for _, pod := range ciliumPods {
+		ciliumServicesRes := kub.CiliumExec(pod, ciliumSvcCmd)
+		err := ciliumServicesRes.GetErr(
+			fmt.Sprintf("Unable to retrieve Cilium services on %s", pod))
+		if err != nil {
+			return err
+		}
+
+		var ciliumSvcs []models.Service
+
+		err = ciliumServicesRes.Unmarshal(&ciliumSvcs)
+		if err != nil {
+			return fmt.Errorf("Unable to unmarshal Cilium services: %s", err.Error())
+		}
+
+		var found bool
+		for _, cSvc := range ciliumSvcs {
+			found = false
+			for _, k8sSvc := range k8sSvcs.Items {
+				if k8sSvc.Spec.ClusterIP == cSvc.Status.Realized.FrontendAddress.IP {
```

Comment:
Can't you invert the logic of those ifs and have a `continue` instead of nesting them?

Reply:
I don't think so, I need to go through all this data in order to validate it, I can factor them out as named functions, but I don't think I can achieve same result without nesting.